src/impl/peerconnection.cpp

```diff
@@ -417,14 +424,17 @@ void PeerConnection::rollbackLocalDescription() {
 
 bool PeerConnection::checkFingerprint(const std::string &fingerprint) const {
 	std::lock_guard lock(mRemoteDescriptionMutex);
-	auto expectedFingerprint = mRemoteDescription ? mRemoteDescription->fingerprint() : nullopt;
-	if (expectedFingerprint && *expectedFingerprint == fingerprint) {
+	std::string expectedFingerprint = "[none]";
+	if (mRemoteDescription && mRemoteDescription->fingerprint().has_value()) {
+		expectedFingerprint = mRemoteDescription->fingerprint().value().value;
+	}
+
+	if (expectedFingerprint == fingerprint) {
```

Comment:
While it should be safe because the description parser checks validity beforehand, comparing the input to "[none]" doesn't feel right to me.
```suggestion
	auto expectedFingerprint = mRemoteDescription && mRemoteDescription->fingerprint() ? mRemoteDescription->fingerprint().value : nullopt;
	if (expectedFingerprint && *expectedFingerprint == fingerprint) {
```